src/bridge/IrcBridge.ts

```diff
@@ -949,6 +958,17 @@ export class IrcBridge {
         );
         for (const [userId, {display_name}] of Object.entries(members)) {
             try {
+                // If the user is banned, skip any connection attempts and go straight for a kick.
+                const banReason = this.matrixBanSyncer?.isUserBanned(userId);
+                if (banReason) {
+                    req.log.debug(`Not syncing ${userId} - user banned (${banReason})`);
+                    this.membershipQueue.leave(
+                        roomId, userId, req, true,
+                        "You are banned from interacting with this bridge.",
```

Comment:
Will they know why?

Reply:
We can use the `reason` field, if we think it's safe to explicitly say.